### PR TITLE
Added a feature to categorize tab as acceptably flaky.

### PR DESCRIPTION
**Pull request by sultan-duisenbay**
Notable changes:

Protos
- Added an option to configure maximum tolerable flakiness to config proto
- Added a flag to denote a tab as acceptably flaky to summary proto
- Added column-wise metrics to summary proto

Code
- Added logic to mark the tab as acceptably flaky &  unit tests for that
- Added validation logic to check if max_acceptable_flakiness is within permitted range (0.0 , 100.0)

### CODE REVIEW

**Review by chases2 — APPROVED** [review]
Overall this looks great, and I'm looking forward to testing it out. Experimental features should be behind a flag, though, so we can disable/enable them for different environments.

/hold

File: pkg/summarizer/summary.go
Comment (chases2) [inline pkg/summarizer/summary.go:699]: Ideally, we would add a flag to gate how statuses are calculated. I think you could do that here:
```
acceptablyFlaky := allowFuzzyFlakiness && acceptableFlakiness(...)
```

... then you could pass in the bool in updateTab (and in update, etc...) through the functions, or with a [package-level variable](https://github.com/GoogleCloudPlatform/testgrid/commit/86f7dd7a44839d93611f6cb4506d064e3d4acc48#diff-6f575e683c5fe1eb9301f314392e63c56a00e7b09ae9663adf4c3e4d67f81c70L786-L788).

See [this commit](https://github.com/GoogleCloudPlatform/testgrid/commit/86f7dd7a44839d93611f6cb4506d064e3d4acc48#diff-b2bac5c2bfa8982ae6b2f98b99cd59f6e8722857209295c2bc74baa83c7bc76fL117-L119) for an example (it's deleting a flag, but it shows what I mean by adding the flag)

**Comment by sultan-duisenbay** [comment]
Makes sense. I will add a flag to cmd/summarizer/main and propagate it all the way down to that function.

File: pkg/summarizer/summary.go
Comment (sultan-duisenbay) [reply pkg/summarizer/summary.go:699]: Makes sense, I will add the flag and propagate it all the way down to this function. Thanks!

Comment (sultan-duisenbay) [reply pkg/summarizer/summary.go]: Done @chases2 & @michelle192837 ! I also added a couple of test cases to TestUpdateTab to see the overall behaviours of this newly added feature.

File: config/config.go
Comment (michelle192837) [inline config/config.go:418]: nit: Note what the valid value should be (e.g. "between 0.0 and 100.0, exclusive.")

File: config/config_test.go
Comment (michelle192837) [inline config/config_test.go:993]: nit: check at the boundaries as well (0.0 and 100.0?)

**Review by michelle192837 — APPROVED** [review]
/hold

Looks good to me aside from nits, feel free to cancel the hold after addressing ^^
(though it might need re-approval, I forget).

File: config/config.go
Comment (sultan-duisenbay) [reply config/config.go:418]: Done

Comment (michelle192837) [reply config/config.go:418]: Ah sorry, I meant in the error message itself, so a user knows what to set the value to. ^^

Comment (michelle192837) [reply config/config.go:418]: Also I misread the logic condition, should be 'inclusive' not 'exclusive'.

**Comment by michelle192837** [comment]
Thanks! `hold /cancel` at will ^^

**Comment by michelle192837** [comment]
(Oh geez how did I typo that >< Thanks!)

**Pull request merged**
